lib/deck.rb

```diff
@@ -0,0 +1,40 @@
+class Deck
+  attr_reader :cards
+
+  def initialize(cards)
+    @cards = cards
+  end
+
+  def rank_of_card_at(index)
+    cards[index].rank
+  end
+
+  def high_ranking_cards
+    high_cards = []
+    cards.each do |card|
+      if card.rank >= 11
+        high_cards << card
+      end
+    end
+    high_cards
+  end
+
+  def percent_high_ranking
+    total_high = 0.0
+    cards.each do |card|
+      if card.rank >= 11
+        total_high += 1
+      end
+    end
```

Comment:
This code is almost identical to the code in your `high_ranking_cards` method. Instead, I would set your `total_high` variable on line 23 equal to `high_ranking_cards` and count that instead.

It would look something like this:
```ruby
  def percent_high_ranking
    total_high = high_ranking_cards.count.to_f
    ((total_high / @cards.length) * 100).round(2)
  end
```